src/windows.toit

```diff
@@ -143,26 +174,62 @@ is-absolute "../foo/bar"       // False.
 ```
 */
 is-absolute path/string -> bool:
+  return is-absolute-volume path or is-absolute-unc path
+
+/**
+Path starts with a UNC volume.
+For example: `//host/share`, `//./UNC/host/share`, ...
+*/
+is-absolute-unc path/string -> bool:
   volume-name-size := volume-name-size_ path
 
   if volume-name-size == 0:
     // Path doesn't have a volume name.
     // For example: `foo/bar`, `../foo`, `\foo\bar`, ...
     return false
 
-  assert: path.size >= 2
-  if is-separator path[0] and is-separator path[1]:
-    // Path starts with a UNC volume.
-    // For example: `//host/share`, `//./UNC/host/share`, ...
-    return true
+  return is-separator path[0]
 
-  if path.size == volume-name-size:
-    // Path is just a volume name.
-    // For example: `c:`.
-    return false
+/**
+Path starts with an absolute drive indicator.
+For example: `c:\\', 'd:/`, ...
+*/
+is-absolute-volume path/string -> bool:
+  return path.size >= 3 and
+      is-volume-letter_ path[0] and
+      path[1] == ':' and
+      is-separator path[2]
 
-  // Make sure it's not just a drive-relative path, like `c:foo/bar`.
-  return is-separator path[volume-name-size]
+/**
+Path starts with a relative drive indicator.
+For example: `c:tmp', 'd:../foo`, 'e:', ...
+*/
+is-relative-volume path/string -> bool:
+  return (path.size == 2 and
+          is-volume-letter_ path[0] and
+          path[1] == ':') or
+         (path.size >= 3 and
+          is-volume-letter_ path[0] and
+          path[1] == ':' and
+          not is-separator path[2])
+
+
+/**
+Is the character $letter a valid volume letter ([a-zA-Z])
+*/
+is-volume-letter_ letter/int -> bool:
+  return 'a' <= letter <= 'z' or 'A' <= letter <= 'Z'
+
+/**
+Returns the volume indicator from an absolute path.
+
+"C:\tmp" -> "C:"
```

Comment:
````suggestion
# Examples
The examples use `/` as `\\` would need to be escaped in the strings.
The results would always return `\\` instead of `/`.
```
volume-indicator "C:/tmp"            // "C:"
volume-indicator "//host/share/foo"  // "//host/share"
```
````